### ISSUE TITLE
Lack of a major version

### ISSUE THREAD

**Issue by jshjohnson** [axios/axios#2265]
> Semver
> Until axios reaches a 1.0 release, breaking changes will be released with a new minor version. For example 0.5.1, and 0.5.4 will have the same API, but 0.6.0 will have breaking changes.

Considering the popularity of Axios and the amount of people using it in production, would it not be prudent to release a major version? The above (taken from the README) is under the heading of `Semver` but it goes against the basis of semantic versioning and is a super risky. If a developer does not read that line and they install axios as `^0.18.1`, they could deploy a breaking change without even realising it. 

I work for a company that is considering transitioning to using `axios` and this issue is potentially going to prevent us doing so.

Thanks

**Comment by jphilipstevens** [comment]
This is a known concern. SEMVER does allow this sort of breaking change in its spec: https://semver.org/#spec-item-4

there seems to be some preliminary effort to make a 1.0 but its not been updated in a while

**Comment by schmod** [comment]
Actually, if you use a specifier like `^0.18.0`, NPM and Yarn generally will _not_ automatically bump you from `0.18.0` to `0.19.0`, even though something like `^1.1.0` _would_ bump you from `1.0.0` to `1.1.0`.  You can see it in action on NPM's [semver calculator](https://semver.npmjs.com/).

That being said, I still consider this to be an issue, because the semantics and behaviors of 0.x releases are generally not well-understood by the community.  

I'd vote for issuing a release that bumps the version to 1.0.0, and Axios can start following normal semver conventions from there on out.  The current status quo is inconvenient, and makes it difficult to keep Axios up to date in a large/complex application.  (I'd imagine that many folks are already using something like Yarn's `resolutions` feature to force Axios to 0.19.0, given the recent flurry of security patches)

Axios is a top-20 NPM package, the [most popular](https://www.npmjs.com/browse/depended) 0.x package in the registry, and the _only_ actively-maintained 0.x package in the top-50.  It hasn't had any major breaking changes in recent memory. It's time to cut a 1.0 release.

tl;dr;  **Let's issue a release that bumps the version to 1.0.0, and does nothing else.**  The current status-quo is an unnecessary inconvenience for a very large number of developers.  A 1.0.0 release would make things easier for a lot of folks at almost no cost.

**Comment by schmod** [comment]
I've opened a [Pull Request](https://github.com/axios/axios/pull/2634).

**Comment by chinesedfan** [comment]
Closed due to https://github.com/axios/axios/issues/1333#issuecomment-619413565.
